### ISSUE TITLE
Docker Desktop fails to add extensions on Fedora

### ISSUE THREAD

**Issue by djnotes** [docker/desktop-linux#15]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID: Could not upload

### Expected behavior
Adding extensions

### Actual behavior
Failing to add extensions 
### Information

  - Linux distro: Fedora 
  - Distro version: 35
  - Docker Desktop Version:
4.8.1 (78998)
### Steps to reproduce the behavior

  1. Install Docker Desktop for Linux pre-requisites (gtk3-devel, pass, docker-ce)
  2. Install Docker Desktop for Linux from the RPM (sudo rpm -i docker-desktop-versionxx.rpm)
  3. Start Docker Desktop and go to Extensions -> Add Extensions and then hit the Install button for one of the listed extensions.
  4. You'll probably see one of these errors:
    - "...unable to find image ..."
    - ".. local image you're trying to install has been modified..." (in case you tried to pull manually with docker cli)
    -"Failed to install extension... due to executing 'docker create image/image:xx no CMD': exit status  1

**Comment by djnotes** [comment]
This issue might be rare, but if you experience the same on Fedora, these steps might be helpful. 

Thanks to [Benjamin Granfond](https://twitter.com/benjaM1), I checked docker context with `docker context ls` and found that DOCKER_HOST environment variable is set to a Podman socket that I had configured before for running docker-compose with Podman. I removed the variable from my `~/.bash_profile` and also ran `unset DOCKER_HOST` in terminal. 

 You also need to change context to desktop-linux so that there is a `*` in front of it. 
```
[mehdi@fedora ~]$ docker context use desktop-linux
desktop-linux
mehdi@fedora ~]$ docker context ls
NAME                TYPE                DESCRIPTION                               DOCKER ENDPOINT                                  KUBERNETES ENDPOINT                     ORCHESTRATOR
default             moby                Current DOCKER_HOST based configuration   unix:///var/run/docker.sock                      https://192.168.39.167:8443 (default)   swarm
desktop-linux *     moby                                                          unix:///home/mehdi/.docker/desktop/docker.sock  
```
DD will not work yet. You need to restart it by pressing the Bug icon at the top and then selecting Restart. However, I didn't know about this and I restarted my computer until it worked.

**Comment by djnotes** [comment]
Up and running!  😃
![Screenshot from 2022-05-11 01-09-13](https://user-images.githubusercontent.com/9320387/167820184-2fb2d7de-66d0-418d-bbca-1906b0d393f3.png)

**Comment by herpall** [comment]
Hello,  i think i have the same problem when i try to install pgadmin in docker desktop i get de following error: "Failed to install extension: the local image you are trying to install has been modified and is different from the published image mochoa/pgadmin4-docker-extension:6.13.0"
